drivers/dma/dma_mcux_lpc.c

```diff
@@ -676,25 +669,18 @@ static int dma_mcux_lpc_init(const struct device *dev)
 {
 	const struct dma_mcux_lpc_config *config = dev->config;
 	struct dma_mcux_lpc_dma_data *data = dev->data;
-	int total_dma_channels;
 
 	/* Indicate that the Otrig Muxes are not connected */
 	for (int i = 0; i < config->num_of_otrigs; i++) {
 		data->otrig_array[i].source_channel = EMPTY_OTRIG;
 		data->otrig_array[i].linked_channel = EMPTY_OTRIG;
 	}
 
-#if defined FSL_FEATURE_DMA_NUMBER_OF_CHANNELS
-	total_dma_channels = FSL_FEATURE_DMA_NUMBER_OF_CHANNELS;
-#else
-	total_dma_channels = FSL_FEATURE_DMA_NUMBER_OF_CHANNELSn(DEV_BASE(dev));
-#endif
-
 	/*
 	 * Initialize to -1 to indicate dma channel does not have a slot
 	 * assigned to store dma channel data
 	 */
-	for (int i = 0; i < total_dma_channels; i++) {
+	for (int i = 0; i < config->num_of_channels; i++) {
```

Comment:
If we are not going to expose a method to the user to allocate configuration data for less than the number of channels present in hardware, this loop (as well as the code here:  https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/dma/dma_mcux_lpc.c#L356) is no longer required, as we don't need a concept of a "virtual channel".

Are you planning to expose a property to set the number of channel configuration structures available in a subsequent PR? Otherwise, we should remove the "virtual channel" management code from the driver.

Reply:
yes I think we should keep the virtual channel stuff for now in case we want to add support for limiting the amount of memory the driver uses based on the channels being used, then we don't have to re-add it, I might do a pr if I have time